### DISCUSSION TITLE
Window handling library

### DISCUSSION THREAD

**Discussion by espkk** [storm-devs/storm-engine#21] [Ideas]
Candidates:

1. SDL2
2. SFML
3. glfw
4. your option

Please, share your options, your analysis of the pros and cons

**Comment by Koncord** [comment]
SDL2 is the industrial standard. So +1 for SDL2

**Comment by ObKo** [comment]
Hello.

I've played a bit with SDL2: https://github.com/ObKo/storm-engine/tree/sdl
For now, only window creation is changed (https://github.com/ObKo/storm-engine/blob/461ad85d35ba1477448d871da87730bf245fb5f7/src/apps/ENGINE/Main.cpp#L144), events are handled same as before, using SDL_SetWindowsMessageHook()

Seems work. Now window has decoration, fullscreen is working properly when there is GUI scaling enabled in system.
